### PR TITLE
Add BuildEngine Properties

### PR DESCRIPTION
**Pull request by chrisvire**

### CODE REVIEW

File: source/OptimaJet.DWKit.StarterApplication/Migrations/20180912174013_AddBuildEngineProperties.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+using Microsoft.EntityFrameworkCore.Migrations;
+
+namespace Optimajet.DWKit.StarterApplication.Migrations
+{
+    public partial class AddBuildEngineProperties : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.AddColumn<string>(
+                name: "PublishingKey",
```
[the remark this answers is not in the document]
Comment (chrisvire) [reply source/OptimaJet.DWKit.StarterApplication/Migrations/20180912174013_AddBuildEngineProperties.cs]: @NullVoxPopuli, at the moment, we only support one publishing key.  It is managed by SAB and is one per machine.

**Comment by NullVoxPopuli** [comment]
If it's one per machine, would it make sense to then support multiple?

**Comment by chrisvire** [comment]
Yes it would, with an infinite amount of resources.

File: source/OptimaJet.DWKit.StarterApplication/Migrations/20180912174013_AddBuildEngineProperties.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                table: "Users",
+                nullable: true);
+
+            migrationBuilder.AddColumn<string>(
+                name: "WorkflowProjectId",
```
[the remark this answers is not in the document]
Comment (chrisvire) [reply source/OptimaJet.DWKit.StarterApplication/Migrations/20180912174013_AddBuildEngineProperties.cs]: After thinking about this more, I might have to move this to Product.  I thought I had worked it all out this morning.

**Comment by chrisvire** [comment]
SAB it not really developed to hand full DVCS.  It is more just using Git for synchronization.

File: source/OptimaJet.DWKit.StarterApplication/Migrations/20180912174013_AddBuildEngineProperties.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (chrisvire) [reply source/OptimaJet.DWKit.StarterApplication/Migrations/20180912174013_AddBuildEngineProperties.cs]: I checked the BuildEngineApi.  We do have support for updating the PublishingKey, but it doesn't maintain which PublishingKeys we have updated.  We could put that in the Scriptoria support.

**Pull request merged**